### ISSUE TITLE
_t types are not POSIX compliant

### ISSUE THREAD

**Issue by ovanes** [zeromq/cppzmq#138]
As of https://stackoverflow.com/a/44573281/98693 POSIX reserved all the names with `_t` suffixes. To be POSIX compliant this lib shouldn't be using any of `_t` suffixes :(

**Comment by bluca** [comment]
Is there any definition actually clashing? Note we can't change public definitions for backward compatibility

**Comment by ovanes** [comment]
yes, this more just for the future knowledge...  Currently there are no clashes, but names like `context_t` or `socket_t` might very likely be used by POSIX. On the other hand, these are C++ classes available in a namespace. Even if there are clashes, one might just need to explicitly put fully qualified namespace before them. Might get a bit murky if they will be macros...
